src/Renderer/Label2DRenderer.js

```diff
@@ -0,0 +1,141 @@
+        this.halfHeight = 0;
+
+        this.transform = '';
+
+        if (document.documentElement.style.transform !== undefined) {
+            this.transform = 'transform';
+        } else if (document.documentElement.style.webkitTransform !== undefined) {
+            this.transform = 'webkitTransform';
+        } else if (document.documentElement.style.mozTransform !== undefined) {
+            this.transform = 'mozTransform';
+        } else if (document.documentElement.style.oTransform !== undefined) {
+            this.transform = 'oTransform';
+        }
+    }
+
+    setSize(width, height) {
+        this.domElement.style.width = `${width}`;
+        this.domElement.style.height = `${height}`;
+
+        this.halfWidth = width / 2;
+        this.halfHeight = height / 2;
+    }
+
+    render(scene, camera) {
+        viewMatrix.copy(camera.matrixWorldInverse);
+        viewProjectionMatrix.multiplyMatrices(camera.projectionMatrix, viewMatrix);
+        cameraSquaredPosition.setFromMatrixPosition(camera.matrixWorld);
+
+        labels = [];
+        this.renderLabels(scene, camera);
+        this.zOrder();
+    }
+
+    renderLabels(object, camera) {
+        // if the object isn't a Label, there are a few options:
+        // - it is visible
+        //      - it has a material but not visible
+        //          - hide its children Label
+        //          - iterate through other children (essentially TileMesh)
+        //      - otherwise iterate through its children
+        // - it not visible, hide all its potential Label children
+        if (!object.isLabel) {
+            if (object.visible) {
+                if (object.material && !object.material.visible) {
+                    object.children.forEach((c) => {
+                        if (c.isLabel) {
+                            c.content.style.display = 'none';
+                        } else {
+                            this.renderLabels(c, camera);
+                        }
+                    });
+                } else {
+                    object.children.forEach(c => this.renderLabels(c, camera));
+                }
+            } else {
+                object.traverse((c) => {
+                    if (c.isLabel) {
+                        c.content.style.display = 'none';
+                    }
+                });
+            }
+
+            return;
+        }
+
+        if (!object.visible) {
+            object.content.style.display = 'none';
+            return;
+        }
+
+        vector.setFromMatrixPosition(object.matrixWorld);
+        vector.applyMatrix4(viewProjectionMatrix);
+
+        if (vector.z < -1 || vector.z > 1) {
+            object.content.style.display = 'none';
+            return;
+        }
+
+        object.x = vector.x;
+        object.y = vector.y;
+
+        // each Label is stored in labels to be sorted in zOrder later
+        labels.push(object);
+
+        if (object.content.parentNode == null) {
```

Comment:
what is this part?

Reply:
The label is attached to the tile, but it hasn't been added to the DOM yet.